js/pages/data-sources/data-sources.js

```diff
@@ -121,9 +123,51 @@ define([
 			this.showSelectionArea = params.showSelectionArea == undefined ? true : params.showSelectionArea;
 			this.currentSource = ko.observable(this.sources[0]);
 			this.currentReport = ko.observable();
+			this.selectedReport = ko.observable();
+
+			this.selectedReportSubscription = this.selectedReport.subscribe(r => {
+				this.updateLocation();
+			});
+
+			this.selectedSourceSubscription = this.currentSource.subscribe(r => {
+				this.updateLocation();
+			})
+
+			this.updateLocation = function () {
```

Comment:
`updateLocation`, `onRouterParamsChange` and `dispose` should be class methods

Reply:
Resolved in https://github.com/OHDSI/Atlas/pull/1165/commits/95e3a3b6ec50ce77c74b79a4227261dbcb89617d